### PR TITLE
Migrate device_dialog to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/device_dialog.dart [hunk 1]
```diff
@@ -38,9 +38,7 @@ class DeviceDialog extends StatelessWidget {
 
     final vm = serviceManager.vm;
 
-    if (vm == null) return const SizedBox();
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/device_dialog.dart:41]: why are we no longer checking for a null vm?

File: packages/devtools_app/lib/src/shared/device_dialog.dart [hunk 2]
```diff
@@ -157,7 +156,7 @@ class _VMFlagsDialogState extends State<VMFlagsDialog> with AutoDisposeMixin {
   }
 
   void _updateFromController() {
-    flags = infoController.flagListNotifier.value.flags
+    flags = infoController.flagListNotifier.value!.flags!
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/device_dialog.dart:159]: we should either make sure flagListNotifier.value cannot return null (maybe empty list instead), or handle a possible null value here e.g. `infoController.flagListNotifier.value?.flags ?? []`

File: packages/devtools_app/lib/src/shared/device_dialog.dart [hunk 3]
```diff
@@ -256,7 +255,7 @@ class _NameColumn extends ColumnData<_DialogFlag> {
         );
 
   @override
-  String getValue(_DialogFlag dataObject) => dataObject.name;
+  String? getValue(_DialogFlag dataObject) => dataObject.name;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/device_dialog.dart:258]: having these ColumnData overrides return null will likely cause issues with the table rendering. I'd set a default value (e.g. '')  instead and have these return types return a non-nullable string

File: packages/devtools_app/lib/src/shared/device_dialog.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/device_dialog.dart:159]: done

File: packages/devtools_app/lib/src/shared/device_dialog.dart (hunk 3 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/device_dialog.dart:258]: done

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/device_dialog.dart]: This comment applies to all of the other getValue / getTooltip overrides too

File: packages/devtools_app/lib/src/shared/device_dialog.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/device_dialog.dart:41]: It should be. It is set to null in vmServiceClosed: https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/shared/service_manager.dart#L332

Comment (polina-c) [reply packages/devtools_app/lib/src/shared/device_dialog.dart:41]: right. ServiceManager is not null safe yet. Thanks for the catch.

File: packages/devtools_app/lib/src/shared/device_dialog.dart (hunk 3 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/device_dialog.dart]: done

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**